src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -297,6 +301,11 @@ private static string MakeNewPath(string originalPath, string newName)
             {
                 if (node is MarkupTagHelperElementSyntax tagHelperElement && BindingContainsTagHelper(originTagHelper, tagHelperElement.TagHelperInfo.BindingResult))
                 {
+                    if (fileRenameUriLookup.TryGetValue(uri, out var newUri))
+                    {
+                        documentIdentifier.Uri = newUri;
+                    }
```

Comment:
Consider moving this to directly after the `documentIdentifier` creation on line 277. It doesn't seem to benefit from being inside the loops.

Alternatively, you could consider doing this as a post-process of the whole `documentChanges` list in `TryGetRazorComponentRenameEditsAsync`, just before the creation of the `WorkspaceEdit`. That approach would mean not having to create a dictionary, pass it around, and look it up for every .razor file in the project.